File: frontend/src/types/message.ts
/**
 * Message-related type definitions.
 */

export interface Message {
  id: string;
  type: 'text' | 'file' | 'system';
  content: string;
  sender: string;
  timestamp: number;
  fileName?: string;
  fileSize?: number;
  fileType?: string;
  downloadUrl?: string;
  fileId?: string;
}

export interface User {
  id: string;
  name: string;
  isOnline: boolean;
}

export interface FileUploadResult {
  success: boolean;
  file_id: string;
  original_name: string;
  content_type: string;
  size: number;
  download_url: string;
  expires_at: string;
}

File: backend/app/models/message.py
"""
Message-related Pydantic models.
"""
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class Message(BaseModel):
    """Message model."""
    id: str
    type: str  # 'text', 'file', 'system'
    content: str
    sender: str
    sender_id: str
    timestamp: datetime
    session_id: str
    fileName: Optional[str] = None
    fileSize: Optional[int] = None
    fileType: Optional[str] = None
    downloadUrl: Optional[str] = None
    fileId: Optional[str] = None

class MessageResponse(BaseModel):
    """Message response model."""
    success: bool
    message: str
    message_id: str

class UserEvent(BaseModel):
    """User event model."""
    type: str  # 'user_joined', 'user_left'
    user_id: str
    user_name: str
    timestamp: datetime
    session_id: str
